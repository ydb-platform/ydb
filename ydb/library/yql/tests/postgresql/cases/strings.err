<sql-statement>
--
-- STRINGS
-- Test various data entry syntaxes.
--
-- SQL string continuation syntax
-- E021-03 character string literals
SELECT 'first line'
' - next line'
	' - third line'
	AS "Three lines to one";
</sql-statement>
<sql-statement>
-- illegal string continuation syntax
SELECT 'first line'
' - next line' /* this comment is not allowed here */
' - third line'
	AS "Illegal comment within continuation";
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:4:1: Error: ERROR:  syntax error at or near "' - third line'"

    	' - third line'
	^
<sql-statement>
-- Unicode escapes
SET standard_conforming_strings TO on;
</sql-statement>
<sql-statement>
SELECT U&'d\0061t\+000061' AS U&"d\0061t\+000061";
</sql-statement>
<sql-statement>
SELECT U&'d!0061t\+000061' UESCAPE '!' AS U&"d*0061t\+000061" UESCAPE '*';
</sql-statement>
<sql-statement>
SELECT U&'a\\b' AS "a\b";
</sql-statement>
<sql-statement>
SELECT U&' \' UESCAPE '!' AS "tricky";
</sql-statement>
<sql-statement>
SELECT 'tricky' AS U&"\" UESCAPE '!';
</sql-statement>
<sql-statement>
SELECT U&'wrong: \061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:19: Error: ERROR:  invalid Unicode escape

    	SELECT U&'wrong: \061';
	                  ^
<sql-statement>
SELECT U&'wrong: \+0061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:19: Error: ERROR:  invalid Unicode escape

    	SELECT U&'wrong: \+0061';
	                  ^
<sql-statement>
SELECT U&'wrong: +0061' UESCAPE +;
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:34: Error: ERROR:  UESCAPE must be followed by a simple string literal at or near "+"

    	SELECT U&'wrong: +0061' UESCAPE +;
	                                 ^
<sql-statement>
SELECT U&'wrong: +0061' UESCAPE '+';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:34: Error: ERROR:  invalid Unicode escape character at or near "'+'"

    	SELECT U&'wrong: +0061' UESCAPE '+';
	                                 ^
<sql-statement>
SELECT U&'wrong: \db99';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair

    	SELECT U&'wrong: \db99';
	                       ^
<sql-statement>
SELECT U&'wrong: \db99xy';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair

    	SELECT U&'wrong: \db99xy';
	                       ^
<sql-statement>
SELECT U&'wrong: \db99\\';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair

    	SELECT U&'wrong: \db99\\';
	                       ^
<sql-statement>
SELECT U&'wrong: \db99\0061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair

    	SELECT U&'wrong: \db99\0061';
	                       ^
<sql-statement>
SELECT U&'wrong: \+00db99\+000061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:27: Error: ERROR:  invalid Unicode surrogate pair

    	SELECT U&'wrong: \+00db99\+000061';
	                          ^
<sql-statement>
SELECT U&'wrong: \+2FFFFF';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:19: Error: ERROR:  invalid Unicode escape value

    	SELECT U&'wrong: \+2FFFFF';
	                  ^
<sql-statement>
-- while we're here, check the same cases in E-style literals
SELECT E'd\u0061t\U00000061' AS "data";
</sql-statement>
<sql-statement>
SELECT E'a\\b' AS "a\b";
</sql-statement>
<sql-statement>
SELECT E'wrong: \u061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:18: Error: ERROR:  invalid Unicode escape

    	SELECT E'wrong: \u061';
	                 ^
<sql-statement>
SELECT E'wrong: \U0061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:18: Error: ERROR:  invalid Unicode escape

    	SELECT E'wrong: \U0061';
	                 ^
<sql-statement>
SELECT E'wrong: \udb99';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair at or near "'"

    	SELECT E'wrong: \udb99';
	                       ^
<sql-statement>
SELECT E'wrong: \udb99xy';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair at or near "x"

    	SELECT E'wrong: \udb99xy';
	                       ^
<sql-statement>
SELECT E'wrong: \udb99\\';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair at or near "\"

    	SELECT E'wrong: \udb99\\';
	                       ^
<sql-statement>
SELECT E'wrong: \udb99\u0061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:24: Error: ERROR:  invalid Unicode surrogate pair at or near "\u0061"

    	SELECT E'wrong: \udb99\u0061';
	                       ^
<sql-statement>
SELECT E'wrong: \U0000db99\U00000061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:28: Error: ERROR:  invalid Unicode surrogate pair at or near "\U00000061"

    	SELECT E'wrong: \U0000db99\U00000061';
	                           ^
<sql-statement>
SELECT E'wrong: \U002FFFFF';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:18: Error: ERROR:  invalid Unicode escape value at or near "\U002FFFFF"

    	SELECT E'wrong: \U002FFFFF';
	                 ^
<sql-statement>
SET standard_conforming_strings TO off;
</sql-statement>
<sql-statement>
SELECT U&'d\0061t\+000061' AS U&"d\0061t\+000061";
</sql-statement>
<sql-statement>
SELECT U&'d!0061t\+000061' UESCAPE '!' AS U&"d*0061t\+000061" UESCAPE '*';
</sql-statement>
<sql-statement>
SELECT U&' \' UESCAPE '!' AS "tricky";
</sql-statement>
<sql-statement>
SELECT 'tricky' AS U&"\" UESCAPE '!';
</sql-statement>
<sql-statement>
SELECT U&'wrong: \061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:19: Error: ERROR:  invalid Unicode escape

    	SELECT U&'wrong: \061';
	                  ^
<sql-statement>
SELECT U&'wrong: \+0061';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:19: Error: ERROR:  invalid Unicode escape

    	SELECT U&'wrong: \+0061';
	                  ^
<sql-statement>
SELECT U&'wrong: +0061' UESCAPE '+';
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:34: Error: ERROR:  invalid Unicode escape character at or near "'+'"

    	SELECT U&'wrong: +0061' UESCAPE '+';
	                                 ^
<sql-statement>
RESET standard_conforming_strings;
</sql-statement>
<sql-statement>
-- bytea
SET bytea_output TO hex;
</sql-statement>
-stdin-:<main>: Error: Parse Sql

    -stdin-:<main>:1:1: Error: VariableSetStmt, not supported name: bytea_output
    	-- bytea
	^
<sql-statement>
SELECT E'\\xDeAdBeEf'::bytea;
</sql-statement>
BackTrace(void**, unsigned long)+29 (0x14087F6D)
TBackTrace::Capture()+30 (0x140884DE)
TWithBackTrace<yexception>::TWithBackTrace<>()+72 (0x14024628)
NYT::TNode::TTypeError::TTypeError()+25 (0x14530469)
NYT::TNode::CheckType(NYT::TNode::EType) const+267 (0x14531A3B)
NYT::TNode::AsString() const+30 (0x145305BE)
WriteTableToStream(IOutputStream&, TVector<NYT::TNode, std::__y1::allocator<NYT::TNode> > const&, TVector<NYT::TNode, std::__y1::allocator<NYT::TNode> > const&)+872 (0x13F0FC28)
Main(int, char**)+5748 (0x13F15084)
main+273 (0x13F16831)
??+0 (0x7F6ED1829D90)
__libc_start_main+128 (0x7F6ED1829E40)
??+0 (0x13EA0029)

(NYT::TNode::TTypeError) library/cpp/yson/node/node.cpp:829: TNode type string_node expected, actual type list_node
