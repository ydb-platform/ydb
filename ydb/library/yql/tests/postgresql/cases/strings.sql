--
-- STRINGS
-- Test various data entry syntaxes.
--
-- SQL string continuation syntax
-- E021-03 character string literals
SELECT 'first line'
' - next line'
	' - third line'
	AS "Three lines to one";
-- illegal string continuation syntax
SELECT 'first line'
' - next line' /* this comment is not allowed here */
' - third line'
	AS "Illegal comment within continuation";
-- Unicode escapes
SET standard_conforming_strings TO on;
SELECT U&'d\0061t\+000061' AS U&"d\0061t\+000061";
SELECT U&'d!0061t\+000061' UESCAPE '!' AS U&"d*0061t\+000061" UESCAPE '*';
SELECT U&'a\\b' AS "a\b";
SELECT U&' \' UESCAPE '!' AS "tricky";
SELECT 'tricky' AS U&"\" UESCAPE '!';
SELECT U&'wrong: \061';
SELECT U&'wrong: \+0061';
SELECT U&'wrong: +0061' UESCAPE +;
SELECT U&'wrong: +0061' UESCAPE '+';
SELECT U&'wrong: \db99';
SELECT U&'wrong: \db99xy';
SELECT U&'wrong: \db99\\';
SELECT U&'wrong: \db99\0061';
SELECT U&'wrong: \+00db99\+000061';
SELECT U&'wrong: \+2FFFFF';
-- while we're here, check the same cases in E-style literals
SELECT E'd\u0061t\U00000061' AS "data";
SELECT E'a\\b' AS "a\b";
SELECT E'wrong: \u061';
SELECT E'wrong: \U0061';
SELECT E'wrong: \udb99';
SELECT E'wrong: \udb99xy';
SELECT E'wrong: \udb99\\';
SELECT E'wrong: \udb99\u0061';
SELECT E'wrong: \U0000db99\U00000061';
SELECT E'wrong: \U002FFFFF';
SET standard_conforming_strings TO off;
SELECT 'tricky' AS U&"\" UESCAPE '!';
RESET standard_conforming_strings;
