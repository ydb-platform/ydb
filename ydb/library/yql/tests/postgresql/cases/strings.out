--
-- STRINGS
-- Test various data entry syntaxes.
--
-- SQL string continuation syntax
-- E021-03 character string literals
SELECT 'first line'
' - next line'
	' - third line'
	AS "Three lines to one";
         Three lines to one          
-------------------------------------
 first line - next line - third line
(1 row)

-- illegal string continuation syntax
SELECT 'first line'
' - next line' /* this comment is not allowed here */
' - third line'
	AS "Illegal comment within continuation";
ERROR:  syntax error at or near "' - third line'"
LINE 3: ' - third line'
        ^
-- Unicode escapes
SET standard_conforming_strings TO on;
SELECT U&'d\0061t\+000061' AS U&"d\0061t\+000061";
 data 
------
 data
(1 row)

SELECT U&'d!0061t\+000061' UESCAPE '!' AS U&"d*0061t\+000061" UESCAPE '*';
 dat\+000061 
-------------
 dat\+000061
(1 row)

SELECT U&'a\\b' AS "a\b";
 a\b 
-----
 a\b
(1 row)

SELECT U&' \' UESCAPE '!' AS "tricky";
 tricky 
--------
  \
(1 row)

SELECT 'tricky' AS U&"\" UESCAPE '!';
   \    
--------
 tricky
(1 row)

SELECT U&'wrong: \061';
ERROR:  invalid Unicode escape
LINE 1: SELECT U&'wrong: \061';
                         ^
HINT:  Unicode escapes must be \XXXX or \+XXXXXX.
SELECT U&'wrong: \+0061';
ERROR:  invalid Unicode escape
LINE 1: SELECT U&'wrong: \+0061';
                         ^
HINT:  Unicode escapes must be \XXXX or \+XXXXXX.
SELECT U&'wrong: +0061' UESCAPE +;
ERROR:  UESCAPE must be followed by a simple string literal at or near "+"
LINE 1: SELECT U&'wrong: +0061' UESCAPE +;
                                        ^
SELECT U&'wrong: +0061' UESCAPE '+';
ERROR:  invalid Unicode escape character at or near "'+'"
LINE 1: SELECT U&'wrong: +0061' UESCAPE '+';
                                        ^
SELECT U&'wrong: \db99';
ERROR:  invalid Unicode surrogate pair
LINE 1: SELECT U&'wrong: \db99';
                              ^
SELECT U&'wrong: \db99xy';
ERROR:  invalid Unicode surrogate pair
LINE 1: SELECT U&'wrong: \db99xy';
                              ^
SELECT U&'wrong: \db99\\';
ERROR:  invalid Unicode surrogate pair
LINE 1: SELECT U&'wrong: \db99\\';
                              ^
SELECT U&'wrong: \db99\0061';
ERROR:  invalid Unicode surrogate pair
LINE 1: SELECT U&'wrong: \db99\0061';
                              ^
SELECT U&'wrong: \+00db99\+000061';
ERROR:  invalid Unicode surrogate pair
LINE 1: SELECT U&'wrong: \+00db99\+000061';
                                 ^
SELECT U&'wrong: \+2FFFFF';
ERROR:  invalid Unicode escape value
LINE 1: SELECT U&'wrong: \+2FFFFF';
                         ^
-- while we're here, check the same cases in E-style literals
SELECT E'd\u0061t\U00000061' AS "data";
 data 
------
 data
(1 row)

SELECT E'a\\b' AS "a\b";
 a\b 
-----
 a\b
(1 row)

SELECT E'wrong: \u061';
ERROR:  invalid Unicode escape
LINE 1: SELECT E'wrong: \u061';
                        ^
HINT:  Unicode escapes must be \uXXXX or \UXXXXXXXX.
SELECT E'wrong: \U0061';
ERROR:  invalid Unicode escape
LINE 1: SELECT E'wrong: \U0061';
                        ^
HINT:  Unicode escapes must be \uXXXX or \UXXXXXXXX.
SELECT E'wrong: \udb99';
ERROR:  invalid Unicode surrogate pair at or near "'"
LINE 1: SELECT E'wrong: \udb99';
                              ^
SELECT E'wrong: \udb99xy';
ERROR:  invalid Unicode surrogate pair at or near "x"
LINE 1: SELECT E'wrong: \udb99xy';
                              ^
SELECT E'wrong: \udb99\\';
ERROR:  invalid Unicode surrogate pair at or near "\"
LINE 1: SELECT E'wrong: \udb99\\';
                              ^
SELECT E'wrong: \udb99\u0061';
ERROR:  invalid Unicode surrogate pair at or near "\u0061"
LINE 1: SELECT E'wrong: \udb99\u0061';
                              ^
SELECT E'wrong: \U0000db99\U00000061';
ERROR:  invalid Unicode surrogate pair at or near "\U00000061"
LINE 1: SELECT E'wrong: \U0000db99\U00000061';
                                  ^
SELECT E'wrong: \U002FFFFF';
ERROR:  invalid Unicode escape value at or near "\U002FFFFF"
LINE 1: SELECT E'wrong: \U002FFFFF';
                        ^
SET standard_conforming_strings TO off;
SELECT 'tricky' AS U&"\" UESCAPE '!';
   \    
--------
 tricky
(1 row)

RESET standard_conforming_strings;
